Issue-2611: Responsive Build Log

AmasiaNalbandian opened the pull request:
<!--
Thanks for sending a pull request!
- if this is your first time, please ensure you have read through our contributor guide: https://github.com/Seneca-CDOT/telescope/blob/master/docs/CONTRIBUTING.md
-->

## Issue This PR Addresses
Fixes #2611 
<!--
1. Automatically close the issue when this PR is merged
    USAGE: Fixes #<issue number>
2. If your PR addresses an issue but does not close it
    USAGE: #<issue number> <reason>(i.e. This issue was worked on by @user and myself and his PR should close the issue.)
-->

## Type of Change

<!-- bug fix, feature, documentation, UI, etc. -->

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description
Make the build log more responsive by using flex.


https://user-images.githubusercontent.com/77639637/152198151-3c9eb3c3-9eb4-4c4f-91ab-cf80e2e7741c.mp4


<!-- Please add a detailed description of what this PR does and why it is needed -->

## Steps to test the PR
Running dashboard front end
1. `pnpm i`
2. `cp config/env.staging .env`
3. `cd ./src/api/status`
4. `pnpm dev`
go to http://localhost:1111/build
 
or go to deployed version and navigate to the build log. You dont need to see a build, to see the changes.
<!-- Please add steps to build the changes in your PR locally so that peers can test your PR
    Example:
    - `npm install`
    - Go to '...'
    - Click on '....'
    - Scroll down to '....'
    - See error

    Please remember, the more clear you are, the faster your PR will be reviewed and merged.
 -->

## Checklist

<!-- Before submitting a PR, address each item -->

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

aserputov commented:
@AmasiaNalbandian I checked this one, and the burger menu isn't working correctly, or should we create an extra issue for that?

aserputov commented:
Also, is that the way it should be 
<img width="1358" alt="Screen Shot 2022-02-02 at 10 35 34 AM" src="https://user-images.githubusercontent.com/65831678/152185889-25f91a78-c3d4-4d00-9aa1-52d24e484d09.png">
?

AmasiaNalbandian commented:
> Also, is that the way it should be. The height is great. <img alt="Screen Shot 2022-02-02 at 10 35 34 AM" width="1358" src="https://user-images.githubusercontent.com/65831678/152185889-25f91a78-c3d4-4d00-9aa1-52d24e484d09.png"> ?

I moved the burger menu on the left since that makes more sense to me. If it controls the left panel, why is it on the right? But how is it not working?

Also that screenshot is not what I had when I fixed this last night...

AmasiaNalbandian commented:
@aserputov Did you build it? Because actually that is what we currently have. I'm not sure the deployment shows the updates for this. Let me quickly attach a screenshot to the description for this PR. That is my bad.

aserputov commented:
@AmasiaNalbandian thanks.

TueeNguyen reviewed (changes requested):
Should the burger be on the right?

Also, let's add some left padding for the terminal, there should be a small gap

![image](https://user-images.githubusercontent.com/58532267/152201652-e9b2be99-7116-4ad5-b458-c2889cf3bf81.png)

DukeManh commented on src/api/status/public/css/terminal.css:
```diff
@@ -15,6 +35,23 @@
+}
+
+.card {
+  border-bottom-right-radius: 0rem !important;
+  border-bottom-left-radius: 0rem !important;
+  border-top-left-radius: 0.75rem !important;
+  border-top-right-radius: 0.75rem !important;
```
```suggestion
    border-radius: top-left top-right botttom-left bottom-right;
```

DukeManh commented on src/api/status/src/views/builds.hbs:
```diff
@@ -29,10 +18,21 @@
-  <div class="container-fluid py-2">
-    <div id="build-header" class="card mb-3 bg-white">
+  <div style="width: 100%; max-height: 100%; ">
+    <div id="build-hewader" class="card bg-white">
```
```suggestion
    <div id="build-header" class="card bg-white">
```

DukeManh commented on src/api/status/src/views/builds.hbs:
```diff
@@ -29,10 +18,21 @@
   </nav>
-  <div class="container-fluid py-2">
-    <div id="build-header" class="card mb-3 bg-white">
+  <div style="width: 100%; max-height: 100%; ">
```
Does `container-fluid` do the same thing as `width: 100%`?

AmasiaNalbandian replied:
container-fluid has other css I did not need, so I added width 100% on its own back since I didnt want the other css.

AmasiaNalbandian commented:
> Should the burger be on the right?
> 
> Also, let's add some left padding for the terminal, there should be a small gap
> 
> ![image](https://user-images.githubusercontent.com/58532267/152201652-e9b2be99-7116-4ad5-b458-c2889cf3bf81.png)
> 
> Other than those I mentioned, everything looks fine

I moved the burger to the left, makes more sense to me since the dashboard display is on the left also. I had no idea that burger on the right would put out a menu on the left, so I made that change in here. I can add the padding, but I believe the dashboard menu has the padding, and thats why it disappears. I'll play around with that.

AmasiaNalbandian commented:
updates were made:

https://user-images.githubusercontent.com/77639637/152281425-1aeb9889-b2a5-48e9-95b7-6150a6ae13b4.mp4

TueeNguyen commented:
@aserputov, we missed this, she didn't rebase all comits

aserputov commented:
@TueeNguyen yes, uhh, how can we redo this ?

humphd commented:
Once it's merged on master, you shouldn't change it.  This is a mistake to correct next time.
